rq/job.py

```diff
@@ -725,4 +735,52 @@ def register_dependency(self, pipeline=None):
             connection.sadd(dependents_key, self.id)
             connection.sadd(self.dependencies_key, dependency_id)
 
+    @property
+    def dependency_ids(self):
+        dependencies = self.connection.smembers(self.dependencies_key)
+        return [Job.key_for(_id.decode())
+                for _id in dependencies]
+
+    def dependencies_are_met(self, exclude_job_id=None, pipeline=None):
+        """Returns a boolean indicating if all of this jobs dependencies are _FINISHED_
+
+        If a pipeline is passed, all dependencies are WATCHed.
+
+        `exclude` allows us to exclude some job id from the status check. This is useful
+        when enqueueing the dependents of a _successful_ job -- that status of
+        `FINISHED` may not be yet set in redis, but said job is indeed _done_ and this
+        method is _called_ in the _stack_ of it's dependents are being enqueued.
+        """
+
+        pipe = pipeline if pipeline is not None else self.connection
+
+        if pipeline is not None:
+            pipe.watch(*self.dependency_ids)
+
+        sort_by = self.redis_job_namespace_prefix + '*->ended_at'
+        get_fields = (
+            '#',
+            self.redis_job_namespace_prefix + '*->created_at',
+            self.redis_job_namespace_prefix + '*->status'
+        )
+
+        # As a minor optimization to more quickly tell if all dependencies
+        # are _FINISHED_, sort dependencies by the `ended_at` timestamp so
+        # those jobs which are not yet finished are at the start of the
+        # list. Sorting here lexographically works because these dates are
+        # stored in an ISO 8601 format, so lexographic order is the same as
+        # chronological order.
+        dependencies_statuses = [
+            tuple(map(as_text, result))
+            for result in pipe.sort(name=self.dependencies_key, by=sort_by,
+                                    get=get_fields, alpha=True,
+                                    groups=True, )
+        ]
```

Comment:
I think we can simplify this bit. We should just fetch all job `statuses` in one pipeline and check if they are all `finished` or `None` (has been deleted).

Reply:
👌 will do --- agreed, this is rather wonky.

Reply:
done in 0b528dae4b763ee8f1f7af9d3a5fd85b301fa6e6 :heavy_check_mark: